examples/hybrid/sphere/held_suarez_rhoe.jl

```diff
@@ -3,35 +3,28 @@ using ClimaCore.DataLayouts
 
 include("baroclinic_wave_utilities.jl")
 
-const sponge = false
+sponge = false
 
-# Variables required for driver.jl (modify as needed)
-horizontal_mesh = cubed_sphere_mesh(; radius = R, h_elem = 4)
-npoly = 4
-z_max = FT(30e3)
-z_elem = 10
-t_end = FT(60 * 60 * 24 * 10)
-dt = FT(400)
-dt_save_to_sol = FT(60 * 60 * 24)
-dt_save_to_disk = FT(0) # 0 means don't save to disk
-ode_algorithm = OrdinaryDiffEq.Rosenbrock23
-jacobian_flags = (; ∂ᶜ𝔼ₜ∂ᶠ𝕄_mode = :no_∂ᶜp∂ᶜK, ∂ᶠ𝕄ₜ∂ᶜρ_mode = :exact)
+setups = [
+    HybridDriverSetup(;
+        additional_cache = make_additional_cache(sponge, true; κ₄ = FT(2e17)),
+        additional_tendency! = make_additional_tendency(sponge, true),
+        center_initial_condition = make_center_initial_condition(:ρe),
+        face_initial_condition = make_face_initial_condition(),
+        horizontal_mesh = cubed_sphere_mesh(; radius = R, h_elem = 4),
+        npoly = 4,
+        z_max = FT(30e3),
+        z_elem = 10,
+        t_end = FT(60 * 60 * 24 * 10),
+        dt = FT(400),
+        dt_save_to_sol = FT(60 * 60 * 24),
+        ode_algorithm = Rosenbrock23,
+        jacobian_flags = (; ∂ᶜ𝔼ₜ∂ᶠ𝕄_mode = :no_∂ᶜp∂ᶜK, ∂ᶠ𝕄ₜ∂ᶜρ_mode = :exact),
+    ),
+]
 
-additional_cache(ᶜlocal_geometry, ᶠlocal_geometry, dt) = merge(
-    hyperdiffusion_cache(ᶜlocal_geometry, ᶠlocal_geometry; κ₄ = FT(2e17)),
-    sponge ? rayleigh_sponge_cache(ᶜlocal_geometry, ᶠlocal_geometry, dt) : (;),
-    held_suarez_cache(ᶜlocal_geometry),
-)
-function additional_tendency!(Yₜ, Y, p, t)
-    hyperdiffusion_tendency!(Yₜ, Y, p, t)
-    sponge && rayleigh_sponge_tendency!(Yₜ, Y, p, t)
-    held_suarez_tendency!(Yₜ, Y, p, t)
-end
-
-center_initial_condition(local_geometry) =
-    center_initial_condition(local_geometry, Val(:ρe))
-
-function postprocessing(sol, output_dir)
+function postprocessing(sols, output_dir)
```

Comment:
sorry, I'm trying to follow. How many `sols` there are? Where are they defined?

Reply:
There used to be a single `sol`, which corresponded to a single "setup", though there did not used to be any proper data structure for storing a setup. Now, the driver gets a list of `setups`, and it generates a list of corresponding `sols`.